File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-16.java
class Solution {
    public int findMin(int[] nums) {
        int len=nums.length;
        int l=0;
        int r=len-1;
        int mid=0;
        if(nums[l]<=nums[r]) return nums[l];
        while(l<r){
            mid=l+(r-l)/2;
            if(nums[mid]>nums[r]){
                l=mid+1;
            }else{
                r=mid;
            }
        }
        return nums[l];
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-16.rs
impl Solution {
    pub fn product_except_self(nums: Vec<i32>) -> Vec<i32> {
        let len=nums.len();
        let mut l_to_r:Vec<i32>=vec![1;len];
        let mut r_to_l:Vec<i32>=vec![1;len];
        let mut res:Vec<i32>=vec![1;len];
        for i in 1..len{
            l_to_r[i]=nums[i-1]*l_to_r[i-1];
        }
        for i in (0..len-1).rev(){
            r_to_l[i]=nums[i+1]*r_to_l[i+1];
        }
        for i in 0..len{
            res[i]=l_to_r[i]*r_to_l[i];
        }
        return res;
    }
}
